### PR TITLE
doc(*): add proposal for enhance orm by nri

### PR DESCRIPTION
**Pull request by Airren**
#### What type of PR is this?

Enhancements

#### What this PR does / why we need it:

Leverage NRI to enhance ORM, thus avoiding intrusive modifications to upstream Kubernetes components. This facilitates synchronous configuration of QoS policies and seamless injection of environment variables

#### Which issue(s) this PR fixes:

https://github.com/kubewharf/katalyst-core/issues/430

### CODE REVIEW

**Comment by caohe** [comment]
@Airren Thanks for your contribution! This proposal is very detailed and well thought out. Could you sign the CLA required by the community?

**Comment by Airren** [comment]
> @Airren Thanks for your contribution! This proposal is very detailed and well thought out. Could you sign the CLA required by the community?

sure, I have finished the CLA signed.

File: docs/proposals/qos-management/orm-nri/20240303-orm-nri.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+spec and returned.
+  ```go
+      func (m *MangerImpl) CreateContainer(pod *api.PodSandbox, container *api.Container) (*api.ContainerAdjustment, []*api.ContainerUpdate, error) {
+        // Update Container Spec from the podResoures
+        adjust, err:= m.updateContainer(pod, container)
+  	    return adjust, nil, err
+      }
+  ```
+
+  **Step 3**, During `RemovePodSandbox()`, all resource allocations related to
```
Comment (caohe) [inline docs/proposals/qos-management/orm-nri/20240303-orm-nri.md:264]: What is the handling mechanism if the `RemovePodSandbox` call fails? We may need to try to ensure that `RemovePodSandbox` is called successfully. If the call fails, the resources allocated to the container may not be released, and the states recorded in the QRM plugin and SysAdvisor may be incorrect.

File: docs/proposals/qos-management/orm-nri/20240303-orm-nri.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+To address the need for intrusive modifications to Kubernetes and changes to the 
+default process, enabling developers to have a more unified implementation 
+approach, NRI has emerged.
+
+[NRI](https://github.com/containerd/nri) is a plugin-based node resource management approach introduced by 
+the upstream community. Using NRI, Kubernetes' node resource management capabilities 
+can be enhanced through plugins without  intrusive modifications to the upstream 
+Kubernetes components.
+
+- NRI allows plugging domain- or vendor-specific custom logic into OCI- compatible
```
[the remark this answers is not in the document]
Comment (Airren) [reply docs/proposals/qos-management/orm-nri/20240303-orm-nri.md]: fixed

File: docs/proposals/qos-management/orm-nri/20240303-orm-nri.md (hunk 1 above)
Comment (Airren) [reply docs/proposals/qos-management/orm-nri/20240303-orm-nri.md:264]: The reconcile mechanism will handle this problem.

**Review by caohe — APPROVED** [review]

**Review by WangZzzhe — APPROVED** [review]

**Pull request merged**